### PR TITLE
DF 831 - return Favourite DTO on create and update + add prefixphrase match 

### PR DESCRIPTION
**Pull request by emina-str**
- return whole DTO in createFavouriteItem and updateFavouriteItem
- refactor creating indexes and fix failing tests
- add IdFixture for FavouriteItem

### CODE REVIEW

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java [hunk 1]
```diff
@@ -38,15 +38,17 @@ public FavouriteItemRestController(FavouriteItemApplicationService favouriteItem
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public FavouriteItemId createFavouriteItem(@RequestBody @Valid CreateFavouriteItemResource createFavouriteItemResource) throws AggregateNotFoundException {
+    public FavouriteItemDto createFavouriteItem(@RequestBody @Valid CreateFavouriteItemResource createFavouriteItemResource) throws AggregateNotFoundException {
         CreateFavouriteItemDto createFavouriteItemDto = new CreateFavouriteItemDto(createFavouriteItemResource.note, createFavouriteItemResource.userId, createFavouriteItemResource.jobAdvertisementId);
-        return favouriteItemApplicationService.create(createFavouriteItemDto);
+        FavouriteItemId favouriteItemId = favouriteItemApplicationService.create(createFavouriteItemDto);
+        return favouriteItemApplicationService.findById(favouriteItemId);
```
Comment (dvoracek-martin) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java]: IMHO not necessary. You can for example return the DTO directly form the service with
FavouriteItemDto favouriteItemDto = FavouriteItemDto.toDto(favouriteItem);

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java [hunk 2]
```diff
@@ -38,15 +38,17 @@ public FavouriteItemRestController(FavouriteItemApplicationService favouriteItem
...
     }
 
     @PutMapping("/{id}/note")
     @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (pavax) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java]: wrong status

Comment (emina-str) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java]: changed to OK, or should I use some other?

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java (hunk 1 above)
Comment (emina-str) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java]: moved to ApplicationService.

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java (hunk 2 above)
Comment (dvoracek-martin) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/FavouriteItemRestController.java]: I disagree - as by other "update" endpoints throughout the application, we should return 204 - NO CONTENT

**Pull request merged**